File: muchos-motors/muchos-motors-api/Models/GenericCarModel.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace muchos_motors_api.Models
{
    public class GenericCarModel
    {
        [Key]
        public int GenericCarModelId { get; set; }
        public string GenerationName { get; set; }
        public string ModelName { get; set; }
        public int CarBrandId { get; set; }
        [ForeignKey("CarBrandId")]
        public CarBrand CarBrand { get; set; }
        public int ProductionStartYear { get; set; }
        public int ProductionEndYear { get; set; }
        public bool IsValid { get; set; }
    }
}

File: muchos-motors-front/src/app/shared/models/GenericCarModel.ts
import { CarBrand } from './CarBrand';

class GenericCarModel {
  genericCarModelId!: number;
  generationName!: string;
  modelName!: string;
  carBrandId!: number;
  carBrand!: CarBrand;
  productionStartYear?: number;
  productionEndYear?: number;
  isValid!: boolean;
}

export default GenericCarModel;
